pkg/ui/src/TextEditable/TextEditable.svelte

```diff
@@ -0,0 +1,60 @@
+<script lang="ts">
+	import { Pencil, Check, X } from 'lucide-svelte';
+
+	/**
+	 * This is the exposed 'value' of the text/input element. This value accepts updates from the parent component, while
+	 * the internal, display value is updated when the input is saved. This results in the following behaviour:
+	 * - parent component binds to this value
+	 * - the parent component updates the 'value' and the udpate is immediately visible in the input
+	 * - the user edits the input - no change is propagated to the parent component
+	 * - the user saves the input - the parent component is updated with the new value
+	 */
+	export let value = '';
+
+	/** This is the internal value, used to store the current state of the input */
+	$: text = value;
+	export let isEditing = false;
+
+	console.log('Is editing: ' + isEditing);
+
+	/** Enter edit mode */
+	function edit() {
+		isEditing = true;
+	}
+	/** Reset the input to the original value */
+	function reset() {
+		text = value;
+		isEditing = false;
+	}
+	/** Save the input and propagate the new value to the parent component */
+	function save() {
+		if (value != text) {
+			value = text;
+		}
+		isEditing = false;
+	}
+</script>
+
+<div {...$$restProps}>
```

Comment:
Not required, just FYI: In Edible's version I ended up spreading `$$restProps` on the input as this component was being used in a form and there were some additional one-off props that I had to pass through from the form state library